src/plugins/blocksmith/components/blocksmith/error-boundary/index.jsx

```diff
@@ -1,75 +1,93 @@
-import { Button } from '@triniti/admin-ui-plugin/components';
 import { connect } from 'react-redux';
 import { ContentBlock, EditorState } from 'draft-js';
-import blockParentNode from '@triniti/cms/plugins/blocksmith/utils/blockParentNode';
-import damUrl from '@triniti/cms/plugins/dam/utils/damUrl';
 import PropTypes from 'prop-types';
 import React from 'react';
+import {
+  Badge,
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+} from '@triniti/admin-ui-plugin/components';
+import blockParentNode from '@triniti/cms/plugins/blocksmith/utils/blockParentNode';
+import createDelegateFactory from '@triniti/app/createDelegateFactory';
+import damUrl from '@triniti/cms/plugins/dam/utils/damUrl';
 import validateBlocks from '@triniti/cms/plugins/blocksmith/utils/validateBlocks';
+
+import { getWordCount } from '../../../utils';
+import delegateFactory from './delegate';
 import selector from './selector';
 import './styles.scss';
 
 const MAX_ERROR_COUNT = 5;
 
-class ErrorBoundary extends React.Component {
+export class ErrorBoundary extends React.Component {
   static propTypes = {
     children: PropTypes.node.isRequired,
+    delegate: PropTypes.shape({
+      handleDirtyEditor: PropTypes.func.isRequired,
+      handleStoreEditor: PropTypes.func.isRequired,
+    }).isRequired,
     editorState: PropTypes.instanceOf(EditorState).isRequired,
     getNode: PropTypes.func.isRequired,
-    onDirtyEditor: PropTypes.func.isRequired,
-    onStoreEditor: PropTypes.func.isRequired,
   };
 
   constructor(props) {
     super(props);
     this.state = {
+      blocks: [],
       errorCount: 0,
       hasError: false,
+      isValid: true,
     };
 
-    this.restoreBlocksmith = this.restoreBlocksmith.bind(this);
+    this.handleRestoreBlocksmith = this.handleRestoreBlocksmith.bind(this);
   }
 
   static getDerivedStateFromError() {
     return { hasError: true };
   }
 
   componentDidCatch(error, errorInfo) {
+    // `setTimeout` ensures that the parent's `setEditorState` fn will always be
+    // invoked first in order to have the latest/greatest prop.
+    setTimeout(() => this.handleRestoreBlocksmith(false));
     console.error(error, errorInfo);
     window.onerror(error);
   }
 
-  restoreBlocksmith(editorState, resetErrorCount = true) {
-    const { onStoreEditor: handleStoreEditor } = this.props;
-    this.setState(({ errorCount }) => ({
+  handleRestoreBlocksmith(resetErrorCount = true, excludeInvalidBlocks = true) {
```

Comment:
@hunyoboy i think `includeInvalidBlocks` would be clearer - what is the benefit of having a negated name for a boolean?

also - im not really into changing this method to not always restore the blocksmith. it's in the name - why not put the early return logic elsewhere? kinda hard to understand as is

Reply:
@efjacobson great eyes. i can't agree more. so because of this - https://github.com/triniti/cms-js/pull/305/commits/e90deaa3537bc0aab6e54cfdb4fcc5351a217a6c